### PR TITLE
Add table aws_guardduty_detector. Closes #203

### LINKED ISSUE #203: Add table aws_guardduty_detector

### PR DESCRIPTION
**Pull request by khushboo9024**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_guardduty_detector []

PRETEST: tests/aws_guardduty_detector

TEST: tests/aws_guardduty_detector
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_partition.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_guardduty_detector.named_test_resource: Creating...
aws_guardduty_detector.named_test_resource: Creation complete after 2s [id=2b08d8d760a14d4bbb5897e2c0bce599]
data.template_file.resource_aka: Refreshing state...

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = 013122550996
aws_partition = aws
region_name = us-east-1
resource_aka = arn:aws:guardduty:us-east-1:013122550996:detector/2b08d8d760a14d4bbb5897e2c0bce599
resource_id = 2b08d8d760a14d4bbb5897e2c0bce599
resource_name = turbottest29507

Running SQL query: test-get-query.sql
[
  {
    "detector_id": "2b08d8d760a14d4bbb5897e2c0bce599",
    "finding_publishing_frequency": "SIX_HOURS",
    "status": "ENABLED",
    "tags": {
      "name": "turbottest29507"
    }
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "akas": [
      "arn:aws:guardduty:us-east-1:013122550996:detector/2b08d8d760a14d4bbb5897e2c0bce599"
    ],
    "detector_id": "2b08d8d760a14d4bbb5897e2c0bce599",
    "status": "ENABLED",
    "tags": {
      "name": "turbottest29507"
    },
    "title": "2b08d8d760a14d4bbb5897e2c0bce599"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "detector_id": "2b08d8d760a14d4bbb5897e2c0bce599",
    "finding_publishing_frequency": "SIX_HOURS",
    "status": "ENABLED"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

POSTTEST: tests/aws_guardduty_detector

TEARDOWN: tests/aws_guardduty_detector

SUMMARY:

1/1 passed.


```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
# Basic info

```sql
select
  detector_id,
  created_at,
  status,
  service_role
from
  aws_guardduty_detector;

+----------------------------------+---------------------+----------+----------------------------------------------------------------------------------------------------------+
| detector_id                      | created_at          | status   | service_role                                                                                             |
+----------------------------------+---------------------+----------+----------------------------------------------------------------------------------------------------------+
| 52b90439c91693313cab4314d2b3fe41 | 2020-05-12 10:34:51 | ENABLED  | arn:aws:iam::013122550996:role/aws-service-role/guardduty.amazonaws.com/AWSServiceRoleForAmazonGuardDuty |
| d7e48832212b40b2b90ee0455922bc71 | 2021-03-18 15:19:53 | DISABLED | arn:aws:iam::013122550996:role/aws-service-role/guardduty.amazonaws.com/AWSServiceRoleForAmazonGuardDuty |
+----------------------------------+---------------------+----------+----------------------------------------------------------------------------------------------------------+
```

```
# List the detectors which are enabled

select
  detector_id,
  created_at,
  status
from
  aws_guardduty_detector
where
  status = 'ENABLED';

+----------------------------------+---------------------+---------+
| detector_id                      | created_at          | status  |
+----------------------------------+---------------------+---------+
| 52b90439c91693313cab4314d2b3fe41 | 2020-05-12 10:34:51 | ENABLED |
+----------------------------------+---------------------+---------+
```

```
# Get finding publishing frequency info for detectors

select
  detector_id,
  status,
  finding_publishing_frequency
from
  aws_new.aws_guardduty_detector

+----------------------------------+----------+------------------------------+
| detector_id                      | status   | finding_publishing_frequency |
+----------------------------------+----------+------------------------------+
| 52b90439c91693313cab4314d2b3fe41 | ENABLED  | SIX_HOURS                    |
| d7e48832212b40b2b90ee0455922bc71 | DISABLED | SIX_HOURS                    |
+----------------------------------+----------+------------------------------+
```
# Get data sources status info for each detectors

```sql
select
  detector_id,
  status as detector_status,
  data_sources -> 'CloudTrail' -> 'Status' as cloud_trail_status,
  data_sources -> 'DNSLogs' -> 'Status' as dns_logs_status,
  data_sources -> 'FlowLogs' -> 'Status' as flow_logs_status
from
  aws_guardduty_detector;
```
```
+----------------------------------+-----------------+--------------------+-----------------+------------------+
| detector_id                      | detector_status | cloud_trail_status | dns_logs_status | flow_logs_status |
+----------------------------------+-----------------+--------------------+-----------------+------------------+
| e8bc3536988d4f5e04bee4852b8ec9c0 | ENABLED         | "ENABLED"          | "ENABLED"       | "ENABLED"        |
+----------------------------------+-----------------+--------------------+-----------------+------------------+
```
```

### CODE REVIEW

File: aws/service.go [hunk 1]
```diff
@@ -318,6 +319,29 @@ func ELBService(ctx context.Context, d *plugin.QueryData, region string) (*elb.E
 	return svc, nil
 }
 
+// GuardDutyService returns the service connection for AWS ELB GuardDuty service
```
Comment (Subhajit97) [inline aws/service.go:322]:
```suggestion
// GuardDutyService returns the service connection for AWS GuardDuty service
```

File: aws/table_aws_guardduty_detector.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+				Type:        proto.ColumnType_TIMESTAMP,
+				Hydrate:     getGuardDutyDetector,
+			},
+			{
+				Name:        "data_sources",
+				Description: "Describes which data sources are enabled for the detector.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getGuardDutyDetector,
+			},
+			// Standard columns
```
Comment (Subhajit97) [inline aws/table_aws_guardduty_detector.go:75]:
```suggestion

			// Standard columns
```

File: aws/table_aws_guardduty_detector.go [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+	}
+
+	return detectorInfo{*op, id}, nil
+}
+
+//// TRANSFORM FUNCTIONS
+
+func getAwsGuardDutyDetectorAkas(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getAwsGuardDutyDetectorAkas")
+	id := h.Item.(detectorInfo)
```
Comment (Subhajit97) [inline aws/table_aws_guardduty_detector.go:174]:
```suggestion
	data := h.Item.(detectorInfo)
```

File: aws/table_aws_guardduty_detector.go [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+	c, err := getCommonColumns(ctx, d, h)
+	if err != nil {
+		return nil, err
+	}
+	commonColumnData := c.(*awsCommonColumnData)
+	aka := "arn:" + commonColumnData.Partition + ":guardduty:" + commonColumnData.Region + ":" + commonColumnData.AccountId + ":detector" + "/" + id.DetectorID
```
Comment (Subhajit97) [inline aws/table_aws_guardduty_detector.go:181]:
```suggestion
	aka := "arn:" + commonColumnData.Partition + ":guardduty:" + commonColumnData.Region + ":" + commonColumnData.AccountId + ":detector/" + data.DetectorID
```

File: aws/table_aws_guardduty_detector.go [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+		logger.Debug("getGuardDutyDetector", "ERROR", err)
+		return nil, err
+	}
+
+	return detectorInfo{*op, id}, nil
+}
+
+//// TRANSFORM FUNCTIONS
+
+func getAwsGuardDutyDetectorAkas(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (Subhajit97) [inline aws/table_aws_guardduty_detector.go:172]: Don't use `get` keyword when defining any transform function

File: docs/tables/aws_guardduty_detector.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+### Get data sources status info for each detectors
+
+```sql
+select
+  detector_id,
+  status as detector_status,
+  data_sources -> 'CloudTrail' -> 'Status' as cloud_trail_status,
+  data_sources -> 'DNSLogs' -> 'Status' as dns_logs_status,
+  data_sources -> 'FlowLogs' -> 'Status' as flow_logs_status
```
Comment (Subhajit97) [inline docs/tables/aws_guardduty_detector.md:42]:
```suggestion
  data_sources -> 'CloudTrail' ->> 'Status' as cloud_trail_status,
  data_sources -> 'DNSLogs' ->> 'Status' as dns_logs_status,
  data_sources -> 'FlowLogs' ->> 'Status' as flow_logs_status
```

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@khushboo9024 There are few changes required. Please see comments.

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**